pkg/restore/restore_test.go

```diff
@@ -792,12 +792,34 @@ func TestInvalidTarballContents(t *testing.T) {
 			)
 
 			assertEmptyResults(t, warnings)
-			assert.Equal(t, tc.wantErrs, errs)
+			assertWantErrs(t, tc.wantErrs, errs)
 			assertAPIContents(t, h, tc.want)
 		})
 	}
 }
 
+func assertWantErrs(t *testing.T, wantErrRes Result, errRes Result) {
+	t.Helper()
+	if wantErrRes.Velero != nil {
+		assert.Equal(t, len(wantErrRes.Velero), len(errRes.Velero))
+		for i := range errRes.Velero {
+			assert.Contains(t, errRes.Velero[i], wantErrRes.Velero[i])
```

Comment:
Should this line use assert.Equal()?

Reply:
Actually, the best was to check the error using assert.ErrorIs. But Result contains strings and not the errors.

After changing the Parser to use the [checkAndReadDir function](https://github.com/vmware-tanzu/velero/blob/bebbd837324eca3a876d4a2b802694e81771e386/pkg/archive/parser.go#L71), I removed the [strings.TrimPrefix](https://github.com/vmware-tanzu/velero/blob/main/pkg/archive/parser.go#L74) of the dir+"/", so in the errors we will get the whole path (include a temp dir that is created as part of the flow). So, we can't use assert.Equal because there will be a different path every time.

Using assert.Contains, We can check that the string of error contains the error we meant to.